theme/static/js/hs_file_browser.js

```diff
@@ -125,6 +126,12 @@ function updateSelectionMenuContext() {
         flagDisableDownload = true;
         flagDisableUnzip = true;
         flagDisableGetLink = true;
+        // Multiple folder deletion is not allowed for composite resource
+        // to avoid race condition that can occur with logical file objects especially
+        // if a folder has many files
+        if(resourceType === 'Composite Resource' && selected.length > 1){
```

Comment:
No, I think this condition could be refined to detect the desired use case (2 or more folders selected) rather than an arbitrarily selection of 2 or more items.